lib/init/rust.js

```diff
@@ -22,45 +22,36 @@ exportables.generateProject = (opts) => {
   options = opts;
 
   return exportables.verifyCargoInstalled()
-    .then(exportables.generateRustSample);
+    .then(exportables.createSampleProgram);
 };
 
-exportables.generateRustSample = () => {
+exportables.createSampleProgram = () => {
   return new Promise((resolve, reject) => {
     // Files, directories, and paths
-    var file_toml = 'Cargo.toml';
-    var file_src = 'main.rs';
-    var dir_src = path.resolve(options.directory, 'src/');
-    var path_toml = path.resolve(options.directory, file_toml);
-    var path_src = path.resolve(dir_src, file_src);
-
-    // Error functions (just to reduce copied text everywhere)
-    function exists_err(filepath) {
-      return new Error(`Looks like this is already a Cargo project! (${filepath} already exists)`);
-    }
-
-    function mkdir_err(dir) {
-      return new Error('Could not create ' + dir);
-    }
+    var cargoToml = 'Cargo.toml';
+    var mainRs = 'main.rs';
+    var srcDir = path.resolve(options.directory, 'src/');
+    var cargoTomlDir = path.resolve(options.directory, cargoToml);
+    var mainRsDir = path.resolve(srcDir, mainRs);
 
     // Generate the toml and the src file
-    fs.exists(dir_src, (exists) => {
+    fs.exists(srcDir, (exists) => {
       if (exists) {
-        return reject(exists_err(dir_src));
+        return reject(new exportables.CargoExistsError(srcDir));
       }
-      fs.exists(path_toml, (exists) => {
+      fs.exists(cargoTomlDir, (exists) => {
         if (exists) {
-          return reject(exists_err(path_toml));
+          return reject(new exportables.CargoExistsError(cargoTomlDir));
         }
-        fs.mkdir(dir_src, (err) => {
-          if (err) {
-            return reject(new Error(mkdir_err(dir_src)));
+        fs.mkdir(srcDir, (error) => {
+          if (error) {
+            return reject(new Error(`Could not create ${srcDir}; ${error.toString()}`));
           }
           // Copy over config file, the blinky main, and the toml file
-          fs.createReadStream(path.resolve(__dirname, './../../resources/rust/', file_toml)).pipe(fs.createWriteStream(path_toml));
+          fs.createReadStream(path.resolve(__dirname, './../../resources/rust/', cargoToml)).pipe(fs.createWriteStream(cargoTomlDir));
           log.info('Initialized Cargo project...');
-          fs.createReadStream(path.resolve(__dirname, './../../resources/rust/', file_src)).pipe(fs.createWriteStream(path_src));
-          log.info(`Wrote "Hello World" to ${path_src}`);
+          fs.createReadStream(path.resolve(__dirname, './../../resources/rust/', mainRs)).pipe(fs.createWriteStream(mainRsDir));
+          log.info(`Wrote "Hello World" to ${mainRsDir}`);
```

Comment:
This one function does the work that several function do in the JS version. For now, I've left it as is, but we should revisit this